src/bindgen/ir/enumeration.rs

```diff
@@ -675,26 +738,31 @@ impl Source for Enum {
                     // so `config.style` is not respected.
                     write!(out, "cdef enum");
                 } else {
-                    write!(out, "{}enum {}", config.style.cython_def(), enum_name);
+                    write!(out, "{}enum {}", config.style.cython_def(), tag_name);
                 }
             }
         }
-
         out.open_brace();
+
+        // Emit enumerators for the tag enum.
         for (i, variant) in self.variants.iter().enumerate() {
             if i != 0 {
                 out.new_line()
             }
             variant.write(config, out);
         }
 
+        // Close the tag enum.
         if config.language == Language::C && size.is_none() && config.style.generate_typedef() {
             out.close_brace(false);
-            write!(out, " {};", enum_name);
+            write!(out, " {};", tag_name);
         } else {
             out.close_brace(true);
         }
 
+        // Emit typedef specifying the tag enum's size if necessary.
+        // In C++ enums can "inherit" from numeric types (`enum E: uint8_t { ... }`),
```

Comment:
I think "inherit" here is a bit weird, I'd just say that they allow specifying the underlying type. But no big deal either way.